metakube/resource_node_deployment_schema.go

```diff
@@ -335,6 +335,17 @@ func matakubeResourceNodeDeploymentCloudOpenstackSchema() map[string]*schema.Sch
 			Description:      "Specifies how long should the controller check if instance is ready before timing out",
 			ValidateDiagFunc: isNonEmptyDurationString,
 		},
+		"server_group_id": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Default:  "",
+			DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+				// if the value is set in the state and unset in the config, the update will not change it
+				// => suppress the diff
+				return oldValue != "" && newValue == ""
```

Comment:
It wouldn't be reverted to the default in that case. If you're not setting the SG ID in the config (anymore, since you removed it in this case), the API will leave it at whatever the current value is (whether that's the default/cluster SG ID or some other SG ID) and return that. So it's correct that it doesn't show a change. The "set to the default SG if it's unset" behaviour only happens at MD creation, not later during updates. If you want to reset it from non-default to default, you have to do that explicitly.

Reply:
got it.
If we wanted to change that (make it defined behavior), we'd have to always send this value client-side and set to default when empty server-side.